### ISSUE TITLE
Documentation on the linked `target()` in `LSP6KeyManager`

### ISSUE THREAD

**Issue by CJ42** [lukso-network/lsp-smart-contracts#138]
I am creating this thread to continue the discussion on the `account()` with @JeneaVranceanu about the `account()` standard function introduced in #131

The idea is to discuss the current implementation, and how potential edge cases can occur.

The `account()` function can return, as one of the options, `ERC725X` only account.
But in the current implementation, it is always considered to be `ERC725Y`.

This function is more intended to be used by external contracts (like the `LSP1UniversalReceiverDelegate`), to fetch the address of the linked account.

The question is _"won't it break if account will refer to a contract that doesn't support ERC725Y?"_

As mentioned by @JeneaVranceanu :
> if the returned account is not supporting ERC725Y then LSP6 is not a key manager at that specific moment/period of time all with that account?

As far as I am aware, the casting in the implementation is used to be able to call the `getData(...)` function.

The KeyManager:

1. receives a payload in raw byte
2. send this payload via a low-level call. 

So since low-level is used, the KeyManager does not check that these functions are present in the linked account.

I would say what matters is actually not about if the linked account supports the standard. But rather if it has one the functions `setData(...)`, `execute(...)` and `transferOwnership(...)`. But we could discuss more of these implications here

**Comment by JeneaVranceanu** [comment]
Thanks for opening this issue.

**TL;DR** I think the account set in the initializer of LSP6KeyManager, the one that is later returned by `function account()`, must support both `ERC725X` and `ERC725Y`.

_For reference: [IERC725X.sol](https://github.com/ERC725Alliance/ERC725/blob/main/implementations/contracts/interfaces/IERC725X.sol)._

My concern was about the following: if "the implementation is used to be able to call the `getData(...)` function" then why `ERC725X`-only account is one of the options that can be returned by `function account()`. 
Judging by the `IERC725x.sol` it has only one function - that is `execute`.

Why is it even an option to set to a KeyManager and return an address that points to a smart contract supporting only `ERC725X` or only `ERC725Y`?

From my point of view, it looks like the address that is set to LSP6KeyManager, which is returned by [`function account()`](https://github.com/lukso-network/lsp-smart-contracts/blob/develop/contracts/LSP6KeyManager/ILSP6KeyManager.sol#L25), must implement both `ERC725X` and `ERC725Y`. 
Why?
1) Key manager directly expects it to support `ERC725Y` so there shouldn't be an option to return `ERC725X`-only account. Of course, anything can be set in the initializer of LSP6KeyManager but it is said in the standard what is the minimum requirement for the linked account and even the solidity docs for the initializer state that it expects that the given address will be the address of `ERC725Account`;
2) LSP6-KeyManager standard expects the support of both ERC725X and ERC725Y. [Link to the standard](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md). See the first screenshot attached below;
3) Judging by the comments the LSP6 interface and its implementation follows the standard (expects ERC725X and ERC725Y interfaces support). See the second screenshot attached below.

**(1) The LSP6 standard summary highlights regarding ERC725X and ERC725Y**
<img width="1180" alt="Screenshot 2022-04-14 at 11 36 21" src="https://user-images.githubusercontent.com/36865532/163347181-40c11f5c-946b-4d4f-8329-f41b8c627584.png">

**(2) What is mentioned in solidity**
<img width="1325" alt="Screenshot 2022-04-14 at 11 33 24" src="https://user-images.githubusercontent.com/36865532/163347221-9d2c2b3e-8f68-4b55-9186-8dafcd46e00a.png">

**Comment by CJ42** [comment]
Thank you for mentioning these different points in the docs and Natspec comments.

@frozeman mentioned that we also want to enable ERC725X only contract to be controlled by a KeyManager in the future. I am thinking of contracts that can act as automated bots (an _"API-like contract"_). These can be created using `ERC725X`, and they can be called only by specific addresses that are allowed to interact with the bot.

The current implementation of the KeyManager contract could work in this way, if the linked account is `ERC725X` only.
As described by @frozeman , the linked `ERC725X` contract could just implement `getData(...)`, so to be able to retrieve permissions stored in its storage, where the storage does not necessarily need to be an `ERC725Y` like contract.

**Comment by JeneaVranceanu** [comment]
Thanks for the explanation! 
Now it makes more sense. It was confusing as the standard wasn't changed but the PR with updated solidity code was merged. Even though the thing that caught my attention was the solidity docs of `function account()` and not the code itself I thought it worth asking "Why?".

As I see there is a pending update to LSP6 in LSPs repo: https://github.com/lukso-network/LIPs/pull/71/files#diff-c38345fd817c0d6f881a5d441fbd1b9ae6aea894805246510557a5059139f887R15

This update adds `account()` function to the standard but with this update the standard summary and `account()` function documentation doesn't match. Probably it should be updated stating e.g. that:
> If the linked account is ERC725X only then it must have `getData()`.

It almost feels like `ERC725Y` should be split into:
- `ERC725Y-Read` for `getData` and 
- `ERC725Y-Write` for `setData` functions.

**Comment by CJ42** [comment]
@JeneaVranceanu this issue might be stale and we might close it.
I have changed the title to something more appropriate. Feel free to comment some suggestions if you consider some improvements can be made.
